libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SyncUpTarget.java

```diff
@@ -144,18 +144,22 @@ public int updateOnServer(SyncManager syncManager, String objectType, String obj
      * @param objectType
      * @param objectId
      * @return
-     * @throws JSONException
-     * @throws IOException
      */
-    public String fetchLastModifiedDate(SyncManager syncManager, String objectType, String objectId) throws JSONException, IOException {
-        final String query = SOQLBuilder.getInstanceWithFields(getModificationDateFieldName())
-                .from(objectType)
-                .where(getIdFieldName() + " = '" + objectId + "'")
-                .build();
-
-        RestResponse lastModResponse = syncManager.sendSyncWithSmartSyncUserAgent(RestRequest.getRequestForQuery(syncManager.apiVersion, query));
-        JSONArray records = lastModResponse.asJSONObject().optJSONArray(Constants.RECORDS);
-        return records != null && records.length() > 0 ? records.optJSONObject(0).optString(getModificationDateFieldName()) : null;
+    public String fetchLastModifiedDate(SyncManager syncManager, String objectType, String objectId) {
```

Comment:
Try-catch
On error we return null which the caller expects - if we throw an exception, the caller will give up on the sync up